bundles/core/src/main/java/com/adobe/cq/wcm/core/components/commons/editor/dialog/inherited/PageImageThumbnail.java

```diff
@@ -95,36 +94,51 @@ protected void initModel() {
         if (currentPage != null) {
             currentPagePath = currentPage.getPath();
         }
-        RequestParameter linkURLParam = request.getRequestParameter(Link.PN_LINK_URL);
-        String linkURL = null;
-        if (linkURLParam != null) {
-            linkURL = linkURLParam.getString();
-        } else {
-            // get the linkURL property defined in the repository for the component
-            ValueMap properties = component.getValueMap();
-            linkURL = properties.get(Link.PN_LINK_URL, String.class);
-        }
 
-        if (StringUtils.isNotEmpty(linkURL)) {
-            targetPage = pageManager.getPage(linkURL);
+        Page targetPage = null;
+        RequestParameter pageLinkParam = request.getRequestParameter("pageLink");
+        if (pageLinkParam != null) {
+
+            // retrieve the page link from the request parameter
+
+            String pageLink = pageLinkParam.getString();
+            targetPage = pageManager.getPage(pageLink);
+
         } else {
-            targetPage = currentPage;
+
+            // retrieve the page link from the component model
+
+            Teaser teaserModel = modelFactory.getModelFromWrappedRequest(request, component, Teaser.class);
+            Link link = null;
+            if (teaserModel != null) {
+                link = teaserModel.getLink();
+            } else {
+                Image imageModel = modelFactory.getModelFromWrappedRequest(request, component, Image.class);
+                if (imageModel != null) {
+                    link = imageModel.getImageLink();
+                }
+            }
+            if (link != null) {
+                targetPage = (Page) link.getReference();
+            } else {
+                targetPage = currentPage;
+            }
         }
 
         if (targetPage == null) {
-            log.error("page is null");
+            log.info("The target page defined for {} is null", component.getPath());
```

Comment:
@jckautzmann - this might not be `component.getPath()`, it could also be due to the `pageLinkParam` not pointing to a valid target page. Perhaps we should account for that in the log message?

Reply:
@richardhand Addressed in https://github.com/adobe/aem-core-wcm-components/pull/1828/commits/484eef1b066e43e55d44e6ffd343f556de31bccb